desktop/cmp/input/NumberInput.ts

```diff
@@ -172,12 +177,18 @@ class NumberInputModel extends HoistInputModel {
         return val;
     }
 
-    override isValid(val: number): boolean {
+    override isValid(val: number | string): boolean {
```

Comment:
This new usage seems to be conflating two different concerns - whether an overall value is valid and whether or not a single initial keypress is valid to begin editing.  

I would like to better understand the job of this particular `isValid` implementation here, in the input model, and a) ensure this code is meeting that contract then b) determine if it does make sense to re-use for the new use case or if we should have a dedicated check there, perhaps a simpler one to confirm that the keystroke is 0-9?

When validating that _first_ keystroke, we aren't expecting any non-numeric characters even if we do have shortcuts or commas enabled. We also don't want to reject e.g. a first keystroke of `1` if a field has a min of `100` - reading the code here, looks like we could have that problem

Reply:
@amcclain Great point about "min of 100".  So, I've rewritten the solution so that it applies only to the NumberEditor.  This solution is inspired by the ["custom numeric editor" example here](https://www.ag-grid.com/react-data-grid/cell-editors/).

I've changed the merge target to `[agGrid31x-reactiveCustomComponents](https://github.com/xh/hoist-react/tree/agGrid31x-reactiveCustomComponents)` since the callback used now is part of agGrid's new
reactiveCustomComponents: true setup.

Reply:
I am ok with this , as it seems to be a mod of ag-Grids own code around this--- thanks for the careful look anselm -- let us know if you are satisfied with the change.